firestore/src/main/java/com/firebase/ui/firestore/paging/FirestorePagingOptions.java

```diff
@@ -66,6 +66,19 @@ public LifecycleOwner getOwner() {
         private LifecycleOwner mOwner;
         private DiffUtil.ItemCallback<DocumentSnapshot> mDiffCallback;
 
+        /**
+         * Directly set the {@link DocumentSnapshot}.
+         * <p>
+         * Do not call this method after calling {@code setQuery}.
+         */
+        @NonNull
+        public Builder<T> setDocumentSnapshot(@NonNull LiveData<PagedList<DocumentSnapshot>> data,
+                                              @NonNull Class<T> modelClass) {
+            mData =  data;
```

Comment:
We should use `Preconditions` to throw if `mData` has already been set, right?

Reply:
How about this try ?